File: tiny_hello_zig/hello.zig
pub export fn _start() callconv(.naked) noreturn {
    asm volatile (
        \\ movq $1, %rax
        \\ movq $1, %rdi
        \\ leaq message(%rip), %rsi
        \\ movq $14, %rdx
        \\ syscall
        \\ movq $60, %rax
        \\ xorq %rdi, %rdi
        \\ syscall
        \\ message:
        \\ .ascii "Hello, world!\n"
    );
    unreachable;
}

File: tiny_hello_c/hello.c
void _start() {
    __asm__ __volatile__(
        "movq $1, %rax\n"              // sys_write
        "movq $1, %rdi\n"              // stdout
        "leaq message(%rip), %rsi\n"
        "movq $14, %rdx\n"
        "syscall\n"

        "movq $60, %rax\n"             // sys_exit
        "xorq %rdi, %rdi\n"
        "syscall\n"

        "message:\n"
        ".ascii \"Hello, world!\\n\"\n"
    );
}
